test/battle/move.c

```diff
@@ -82,55 +82,105 @@ SINGLE_BATTLE_TEST("Turn order is determined randomly if priority and Speed tie
+                break;
+            case 235360:
+                gBattleTestRunnerState->occurenceCounter += 8;
+                break;
+            case 262360:
+                gBattleTestRunnerState->occurenceCounter += 16;
+                break;
+            case 202360:
+                gBattleTestRunnerState->occurenceCounter += 32;
+                break;
+            case 189378:
+                gBattleTestRunnerState->occurenceCounter += 64;
+                break;
+            case 189189:
+                gBattleTestRunnerState->occurenceCounter += 128;
+                break;
+            case 189480:
+                gBattleTestRunnerState->occurenceCounter += 256;
+                break;
+            case 188480:
+                gBattleTestRunnerState->occurenceCounter += 512;
+                break;
+            case 188240:
+                gBattleTestRunnerState->occurenceCounter += 1024;
+                break;
+            case 188188:
+                gBattleTestRunnerState->occurenceCounter += 2048;
+                break;
+            case 262262:
+                gBattleTestRunnerState->occurenceCounter += 4096;
+                break;
+            case 262142:
+                gBattleTestRunnerState->occurenceCounter += 8192;
+                break;
+            case 202403:
+                gBattleTestRunnerState->occurenceCounter += 16384;
+                break;
+            case 202202:
+                gBattleTestRunnerState->occurenceCounter += 32768;
+                break;
+            case 262283:
+                gBattleTestRunnerState->occurenceCounter += 65536;
+                break;
+            case 202283:
+                gBattleTestRunnerState->occurenceCounter += 131072;
+                break;
+            case 235180:
+                gBattleTestRunnerState->occurenceCounter += 262144;
+                break;
+            case 261180:
+                gBattleTestRunnerState->occurenceCounter += 524288;
+                break;
+            case 235235:
+                gBattleTestRunnerState->occurenceCounter += 1048576;
+                break;
+            case 235300:
+                gBattleTestRunnerState->occurenceCounter += 2097152;
+                break;
+            case 261141:
+                gBattleTestRunnerState->occurenceCounter += 4194304;
+                break;
+            case 261261:
+                gBattleTestRunnerState->occurenceCounter += 8388608;
+                break;
+        }
+        EXPECT_EQ(gBattleTestRunnerState->occurenceCounter, 16777215);
+        //  Any of the following pairs of hp values work:
+        //  (188, 360), (189, 360), (261, 360), (235, 360), (262, 360), (202, 360),
+        //  (189, 378), (189, 189), (189, 480), (188, 480), (188, 240), (188, 188),
+        //  (262, 262), (262, 142), (202, 403), (202, 202), (262, 283), (202, 283),
+        //  (235, 180), (261, 180), (235, 235), (235, 300), (261, 141), (261, 261)
```

Comment:
I think we don't need this comment anymore? Or if you'd like to keep it, maybe put it before the `switch`?

Reply:
I can put an explaining comment before the switch, the hp pairs are baked into the switch now so it's not really needed. Along with the explanation for how the switch works.